osu.Game/Overlays/Chat/ChatLine.cs

```diff
@@ -200,13 +213,21 @@ protected override void LoadComplete()
             FinishTransforms(true);
         }
 
+        /// <summary>
+        /// Schedules a message highlight animation.
+        /// </summary>
+        /// <remarks>
+        /// Scheduling is required to ensure the animation doesn't play until the chat line is in view and not scrolled away.
+        /// </remarks>
+        public void ScheduleHighlight() => Schedule(() => lineBackground.FlashColour(usernameColour.Darken(1f).Opacity(0.5f), 1000, Easing.InQuint));
```

Comment:
Nitpick but technically the more proper way to do this fade would be to do something the lines of

```suggestion
        public void ScheduleHighlight() => Schedule(() =>
        {
            var backgroundColour = usernameColour.Darken(1f);
            lineBackground.Colour = backgroundColour.Opacity(0);
            lineBackground.FlashColour(backgroundColour.Opacity(0.5f), 1000, Easing.InQuint);
        });
```

The flash as written in this PR will interpolate from the darkened-transparent `usernameColour` to fully transparent black, which means not only the alpha will change, the colour will actually fade to black at the same time as well. Setting `usernameColour` to have all components other than alpha the same fixes this, as the colour part will not interpolate, just the alpha will.

In fact this could be vastly simplified by assigning `lineBackground.Colour = usernameColour` and only modulating alpha via `FadeTo()` rather than doing these colour manipulations like here. Unless the darkening was somehow explicitly intended?

Reply:
Can simplify by assigning the username colour to `lineBackground` directly as you mentioned, didn't do that as I was initially using a constant highlight colour, where it doesn't make sense for `lineBackground` to have such colour assigned to it.